app/src/main/webapp/js/controllers.js

```diff
@@ -1,581 +1,553 @@
+		        }
+		    );
+		    $scope.message = null;	
+		    $scope.messageSelections = [];
+		
+		    // ngGrid
+		    $scope.$on(
+		        'ngGridEventColumns', 
+		        function(event, columns) {
+		            errorColumnPrefs.save(columns);
+		        }
+		    );
+		
+		    var columnDefs = errorColumnPrefs.load();
+		
+		    $scope.sortOptions = {
+		        fields : [ "timestamp" ],
+		        directions : [ "asc" ]
+		    };
+		
+		    $scope.totalServerItems = 0;
+		
+		    $scope.pagingOptions = {
+		        pageSizes : [ 12, 20, 50, 100 ],
+		        pageSize : 12,
+		        currentPage : 1
+		    };
+		
+		    $scope.$on(
+		    	'errorGridResize', 
+		    	function() {
+		    		layoutPlugin.updateGridLayout();
+		    	}
+		    );
+		
+		    $scope.gridOptions = {
+		        plugins: [layoutPlugin],
+		        data : 'messages',
+		        columnDefs : columnDefs,
+		        enablePaging : true,
+		        showFooter : true,
+		        totalServerItems : 'totalServerItems',
+		        pagingOptions : $scope.pagingOptions,
+		        selectedItems : $scope.messageSelections,
+		        multiSelect : false,
+		        useExternalSorting : true,
+		        sortInfo : $scope.sortOptions,
+		        showColumnMenu: true,
+		        enableColumnReordering: true,
+		    };
+		
+		    var parseSearchString = function(searchStr) {
+		        var searchParams = searchStr.split(';');
+		        var criteria = {};
+		        searchParams.forEach(function(param) {
+		        	var keyValue = param.split('=');
+		            if (keyValue.length == 2) {
+		            	criteria[keyValue[0].trim()] = keyValue[1]
+		                .replace(/"/g, '')
+		                .trim();
+		            } else {
+		                if (param) {
+		                	$log.error("Could not parse: "+ param);
+		                }
+		            }
+		          }
+		        );
+		        return criteria;
+		    };
+		
+		    var prepareResultsPage = function() {
+		        var first = ($scope.pagingOptions.currentPage - 1) * $scope.pagingOptions.pageSize;
+		        var maxResults = $scope.pagingOptions.pageSize;
+		        var sortField = $scope.sortOptions.fields[0];
+		        var sortAsc = ($scope.sortOptions.directions[0] === "asc");
+		
+		        if ($rootScope.searchField_searchStr) {
+		            var criteria = parseSearchString($rootScope.searchField_searchStr);
+		
+		            EsbMessageService.search(
+		                criteria, 
+		                $scope.fromDate,
+		                $scope.toDate, 
+		                first,
+		                maxResults, 
+		                sortField,
+		                sortAsc
+		            ).then(
+		                function(response) {
+		                	$scope.messages = response.data.messages;
+		                	$scope.totalServerItems = response.data.totalResults;
+		                                
+		                    if ($scope.totalServerItems === 0) {
+		                    	$scope.message = null;
```

Comment:
Looks like this is the problem. If you change this to `$scope.messageSelections.splice(0, $scope.messageSelections.length);`, it will work. Splice alters the contents of an array. Specifically, this deletes all of the elements in an array. We would not want to reassign an empty array, because then `messageSelections` would be assigned a new, different array than what our ng-grid is acting on (we pass it `$scope.messageSelections` in `gridOptions` -- this is a pointer to a specific array so we must modify this array and not assign a new one).

Some "why" and some suggestions to prevent this kind of confusion in the future:

## why?
Why does `$scope.message = null` not reset the selection on the UI as expected? Because there is a hierarchy of scopes, and the way JavaScript works with object properties.

There are two controllers at play: ErrorsSearchCtrl and ErrorsDetailCtrl. I'll just refer to them as SearchCtrl and DetailCtrl. All controllers get their own scope, though they can be nested. Nesting happens simply by virtue of that controller being used in a child element in the DOM. Therefore, DetailCtrl is nested underneath SearchCtrl (refer to errors.html). If a value is `undefined` in a controller's scope, it will fall back to its parent scope's value, and so on until a value is found, or there are no more parent scopes to look for. Given that, let's walk through what these controller's scopes look like and what expressions will resolve to on that scope step by step:

### 1. page loads
| controller | scope.message actual value | `{{message}}` within this scope      |
| ------------ | ------------------------------------- | --------------------------------------------- |
| Search     | null                                        | null                                                 |
| Details     | undefined                              | null (uses parent)                           |

### 2. Search and get messages, then select a message
| controller | scope.message actual value | `{{message}}` within this scope  |
| ------------ | ------------------------------------- | --------------------------------------------- |
| Search     | null                                        | null                                                 |
| Details     | {...message object...}             | {...message object...}                     |

Why does Details scope.message get set and not Search's scope.message? Because in DetailCtrl: `$scope.message = ...` (line 200) will assign the object to _its_ scope's message property.

### 3. Search, get no messages, selection is not reset on UI...
| controller | scope.message actual value | `{{message}}` within this scope  |
| ------------ | ------------------------------------- | --------------------------------------------- |
| Search     | null                                        | null                                                 |
| Details     | {...message object...}             | {...message object...}                     |

Why? In SearchCtrl, when it get's a response with no messages, it sets _its_ scope.message to null (which it was already actually). But the DetailCtrl child scope still has a value, so it ignores it. Therefore, the UI does not get updated.

## suggestions
I think this would be clearer if you did not use two controllers (#37), since obviously you would then not have two scopes acting in hierarchy.

Additionally, if you combine them, I think you can consider getting rid of the assignment and use of `$scope.selectedMessage` and just update `$scope.message` when `$scope.messageSelections` changes. This makes one less scope property to keep track of in your head.

I opened a PR for a fix and a test to reproduce this issue. Once I got the test setup (which was a good exercise in and of itself), it was very easy to debug the issue. With these kinds of tricky issues, reproducing it with a test is a good place to start IMHO.